flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/handlers/utils/JarHandlerUtils.java

```diff
@@ -189,7 +189,11 @@ public PackagedProgram toPackagedProgram(Configuration configuration) {
                         .setArguments(programArgs.toArray(new String[0]))
                         .build();
             } catch (final ProgramInvocationException e) {
-                throw new CompletionException(e);
+                throw new CompletionException(
```

Comment:
I'm not sure whether we assume here that the `ProgramInvocationException` is caused by something the user provided (which would allow the BAD_REQUEST response). [PackagedProgram#extractContainedLibraries](https://github.com/apache/flink/blob/c08bef1f7913eb1c416c278354fd62b82b172549/flink-clients/src/main/java/org/apache/flink/client/program/PackagedProgram.java#L554) throws a `ProgramInvocationException` for unknown IO errors, for instance, which would be actually an internal issue, wouldn't it?

Reply:
Thanks @XComp for review.

We have a tool that uses a REST API to submit jobs, but if the user provides the wrong entry class name by mistake, Flink returns an INTERNAL_SERVER_ERROR, which can be confusing.

You bring up a valid concern regarding exceptions like those thrown by `PackagedProgram#extractContainedLibrarie`s, which might be related to internal issues. To address this, we should consider improving our error handling to distinguish between user-provided errors and internal issues. This way, we can ensure that our responses accurately reflect the nature of the problem and provide better feedback to our users.

I will try to find a better solution for this. Do you have any suggestions?

Reply:
> We have a tool that uses a REST API to submit jobs, but if the user provides the wrong entry class name by mistake, Flink returns an INTERNAL_SERVER_ERROR, which can be confusing.

I'm not questioning your intention. I just want to point out that the approach you follow might not be sufficient. The error handling of the job submission is not that concise. It requires a bigger effort to make this right in my opinion.

I'm happy to help if you're interested in addressing this issue on a broader angle.